net.go

```diff
@@ -90,7 +91,19 @@ func Dial(network, addr string, ctx *Ctx, flags DialFlags) (*Conn, error) {
 func DialTimeout(network, addr string, timeout time.Duration, ctx *Ctx,
 	flags DialFlags) (*Conn, error) {
 	d := net.Dialer{Timeout: timeout}
-	return dialSession(d, network, addr, ctx, flags, nil)
+	return dialSession(d, context.Background(), network, addr, ctx, flags, nil)
+}
+
+// DialContext acts like Dial but takes a context for network dial.
+//
+// The context includes only network dial. It does not include OpenSSL calls.
+//
+// See func Dial for a description of the network, addr, ctx and flags
+// parameters.
+func DialContext(context context.Context, network, addr string,
```

Comment:
It would be nice to see some simple test for the call if it's not too complicated.

Reply:
New tests added into `net_test.go`.